capa/rules/cache.py

```diff
@@ -107,6 +161,41 @@ def get_ruleset_content(ruleset: capa.rules.RuleSet) -> List[bytes]:
 
 def compute_ruleset_cache_identifier(ruleset: capa.rules.RuleSet) -> CacheIdentifier:
     rule_contents = get_ruleset_content(ruleset)
+
+    if is_dev_environment():
+        modified_files = get_modified_files()
```

Comment:
can we get away with ensuring that only if the rules related files have been modified we factor that into the hash and otherwise bail?

Reply:
i.e. only if the ruleset/cache affecting source file has been modified we generate a new cache file

Reply:
>can we get away with ensuring that only if the rules related files have been modified we factor that into the hash and otherwise bail?

This could be a potential improvement, though I need to look further into it. Currently, we generate a new cache if at least one of the following conditions is met: both a rule file (.yml) and a source file (.py) have been modified (or just at least one of them), or a new commit hash.